components/eventing-controller/pkg/env/nats_config.go

```diff
@@ -46,5 +55,18 @@ func GetNatsConfig(maxReconnects int, reconnectWait time.Duration) NatsConfig {
 	if err := envconfig.Process("", &cfg); err != nil {
 		log.Fatalf("Invalid configuration: %v", err)
 	}
+
+	// set stream name for JetStream
+	if len(cfg.EventTypePrefix) > 0 {
+		cfg.JSStreamName = getStreamNameForJetStream(cfg.EventTypePrefix)
+		if len(cfg.JSStreamName) == 0 {
+			log.Fatalf("Invalid configuration! Failed to extract appropriate stream name from EventTypePrefix: %s", cfg.EventTypePrefix)
+		}
+	}
 	return cfg
 }
+
+func getStreamNameForJetStream(eventTypePrefix string) string {
+	streamName := strings.ToLower(strings.Split(eventTypePrefix, ".")[0])
```

Comment:
Why do we need `ToLower`? Is this a constraint by JetStream?

Reply:
1. I still believe `ToLower` is not needed here because it doesn't have any effect regarding the naming collision. But feel free to leave it like that.
2. So `a.b.c` becomes `a`, but why can't it become `abc`.
3. We should introduce a logic to limit the stream name to 32 characters.

Reply:
2. If it becomes `abc`, then the events filter will be `abc.>`. It means then for the JetStream subscriptions (suppose eventType: a.b.c.XYZ) we need to append again as `abc.a.b.c.XYZ`.

Reply:
This conversation is not relevant anymore.